### PR TITLE
Data addresses #minor

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
Add a new endpoint to return friendlier names for Literal Map inputs and outputs.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

### CODE REVIEW

File: protos/flyteidl/service/admin.proto [hunk 1]
```diff
@@ -19,8 +19,6 @@ import "flyteidl/admin/task_execution.proto";
 import "flyteidl/admin/version.proto";
 import "flyteidl/admin/common.proto";
 import "flyteidl/admin/description_entity.proto";
-import "flyteidl/core/identifier.proto";
```
Comment (wild-endeavor) [inline protos/flyteidl/service/admin.proto:22]: just cleanup from the output of `make generate`

**Review by EngHabu — COMMENTED** [review]
Awesome!

File: protos/flyteidl/service/dataproxy.proto [hunk 2]
```diff
@@ -102,6 +104,29 @@ message CreateDownloadLinkResponse {
...
+  string flyte_url = 1;
+}
+
+// General request artifact to retrieve data from a Flyte artifact url.
+message GetDataRequest {
+  FlyteArtifact artifact = 1;
+}
+
+message GetDataResponse {
+  oneof data {
```
Comment (EngHabu) [inline protos/flyteidl/service/dataproxy.proto:121]: Should we also return information about the link (node exec id, type of output... etc.)

File: protos/flyteidl/service/dataproxy.proto [hunk 3]
```diff
@@ -102,6 +104,29 @@ message CreateDownloadLinkResponse {
...
+    // literal map data will be returned
+    core.LiteralMap literal_map                = 1;
+
+    // Flyte deck html will be returned as a signed url users can download
+    CreateDownloadLinkResponse flyte_deck_download_link = 2;
```
Comment (kumare3) [inline protos/flyteidl/service/dataproxy.proto:126]: embedding one response in another seems like a code-smell

File: protos/flyteidl/service/dataproxy.proto [hunk 4]
```diff
@@ -136,4 +161,11 @@ service DataProxyService {
     //   description: "Creates a read-only http location that is accessible for tasks at runtime."
     // };
   }
+
+  rpc GetData (GetDataRequest) returns (GetDataResponse) {
+    // Takes an address like flyte://v1/proj/development/execid/n2/0/i and return the actual data
+    option (google.api.http) = {
+      get: "/api/v1/data/get_artifact"
```
Comment (kumare3) [inline protos/flyteidl/service/dataproxy.proto:168]: why do we need `get_artifact`
this is already a `GET` api
I would keep it as
`GET /api/v1/data?{data-uri}`

File: protos/flyteidl/service/dataproxy.proto [hunk 5]
```diff
@@ -102,6 +104,29 @@ message CreateDownloadLinkResponse {
...
+// backend, identifies a Flyte artifact (input, output, flyte deck, etc.).
+// e.g. flyte://v1/proj/development/execid/n2/0/i (for 0th task execution attempt)
+//      flyte://v1/proj/development/execid/n2/i (for node execution)
+message FlyteArtifact {
+  string flyte_url = 1;
+}
+
+// General request artifact to retrieve data from a Flyte artifact url.
+message FlyteArtifactGetRequest {
+  FlyteArtifact artifact = 1;
```
[the remark this answers is not in the document]
Comment (wild-endeavor) [reply protos/flyteidl/service/dataproxy.proto:117]: yes

File: protos/flyteidl/service/dataproxy.proto (hunk 3 above)
Comment (kumare3) [reply protos/flyteidl/service/dataproxy.proto:122]: oh man i forget did we end up deciding on using dataproxy? if so i guess +1. Also response inside response is really not loved. it confuses the client user, but i see there is no wrapper inside it. 

By the way should we also add link to the checkpoint if it is available - like in the task node case - https://github.com/flyteorg/flyteidl/blob/c7ae927f73df9ecdbbf01b6021836b9b2079e527/protos/flyteidl/event/event.proto#L128? I do not think it should be signed.

File: protos/flyteidl/service/dataproxy.proto [hunk 6]
```diff
@@ -136,4 +157,11 @@ service DataProxyService {
     //   description: "Creates a read-only http location that is accessible for tasks at runtime."
     // };
   }
+
+  rpc GetData (GetDataRequest) returns (GetDataResponse) {
+    // Takes an address like flyte://v1/proj/development/execid/n2/0/i and return the actual data
+    option (google.api.http) = {
+      get: "/api/v1/data"
```
Comment (kumare3) [inline protos/flyteidl/service/dataproxy.proto:164]: can we change this
```suggestion
      get: "/api/v1/data?{address}"
```

Comment (wild-endeavor) [reply protos/flyteidl/service/dataproxy.proto:164]: is that how you do it?  i don't think you need this for grpc gateway.  this already works like this.  like what i'm sending in postman is
```
http://localhost:8088/api/v1/data?flyte_url=flyte://v1/flytesnacks/development/f6988c7bdad554a4da7a/n0/d
```

File: protos/flyteidl/service/dataproxy.proto (hunk 3 above)
Comment (EngHabu) [reply protos/flyteidl/service/dataproxy.proto:122]: I think a separate object sounds better...

**Review by EngHabu — APPROVED** [review]

**Review by eapolinario — APPROVED** [review]

**Review by pingsutw — APPROVED** [review]

**Review by EngHabu — APPROVED** [review]

**Review by pingsutw — APPROVED** [review]

**Pull request merged**